inspirehep/modules/orcid/putcode_getter.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2018 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+from __future__ import absolute_import, division, print_function
+
+import itertools
+import re
+import six
+from flask import current_app
+
+from inspire_services.orcid.client import OrcidClient
+from inspire_services.orcid import exceptions as orcid_client_exceptions
+
+from . import exceptions
+
+
+INSPIRE_WORK_URL_REGEX = re.compile(
+    r'https?://(?:labs\.)?inspirehep\.net/(?:record|literature)/(\d+)',
+    re.IGNORECASE
+)
+
+
+class OrcidPutcodeGetter(object):
+    def __init__(self, orcid, oauth_token):
+        self.orcid = orcid
+        self.oauth_token = oauth_token
+        self.client = OrcidClient(self.oauth_token, self.orcid)
+        self.source_client_id_path = current_app.config['ORCID_APP_CREDENTIALS'][
+            'consumer_key']
+
+    def get_all_inspire_putcodes(self):
+        self._get_all_putcodes()
+        if not self.putcodes:
+            return
+        # Filter out putcodes that do not belong to Inspire.
+        for putcode, url in self._get_urls_for_all_putcodes():
+            if INSPIRE_WORK_URL_REGEX.match(url):
+                yield putcode, url
+
+    def _get_all_putcodes(self):
```

Comment:
Feels like a getter should return something, but instead it stores it as instance state. How about returning instead, or maybe renaming to `refresh` or similar?

Reply:
Very good point, it should return